pkg/util/requestbuffers.go

```diff
@@ -3,19 +3,26 @@ package util
 
 import (
 	"bytes"
-	"sync"
 )
 
+// Pool is an abstraction of sync.Pool, for testability.
+type Pool interface {
```

Comment:
this looks like an existing interface, which made me think this is probably a good fit for that pool package, but that's a different topic

https://github.com/grafana/mimir/blob/942e16cfc8699ed2ea35aad439d05c730560a556/pkg/util/pool/pool.go#L12-L18

Reply:
Aha, thanks @dimitarvdimitrov - I was not aware of this package.